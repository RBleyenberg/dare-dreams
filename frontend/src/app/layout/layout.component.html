<mat-sidenav-container class="sidenav-container">
  <mat-sidenav
    #drawer
    class="sidenav"
    fixedInViewport="true"
    [attr.role]="(isHandset$ | async) ? 'dialog' : 'navigation'"
    [mode]="(isHandset$ | async) ? 'over' : 'side'"
    [opened]="!(isHandset$ | async)">
    <mat-toolbar color="primary">Menu</mat-toolbar>
    <mat-nav-list>
      <a mat-list-item routerLink="articles">Artikelen</a>
      <a mat-list-item routerLink="relations">Relaties</a>
      <a mat-list-item routerLink="relatie-type">Relatie type</a>
      <a mat-list-item routerLink="valutas">Valuta's</a>
    </mat-nav-list>
  </mat-sidenav>
  <mat-sidenav-content>
    <mat-toolbar color="primary">
      <button
        type="button"
        aria-label="Toggle sidenav"
        mat-icon-button
        (click)="drawer.toggle()"
        *ngIf="isHandset$ | async">
        <mat-icon aria-label="Side nav toggle icon">menu</mat-icon>
      </button>
      <span>frontend</span>
    </mat-toolbar>
<ng-content>

</ng-content>
  </mat-sidenav-content>
</mat-sidenav-container>
