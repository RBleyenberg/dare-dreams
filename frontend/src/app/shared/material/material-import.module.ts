import {
  MatButtonModule, MatCardModule,
  MatCheckboxModule, MatChipsModule,
  MatIconModule, MatInputModule,
  MatListModule, MatMenuModule, MatProgressSpinnerModule, MatRadioModule, MatSelectModule,
  MatSidenavModule, MatSlideToggleModule, MatSnackBarModule, MatTabsModule,
  MatToolbarModule, MatTooltipModule
} from '@angular/material';
import {NgModule} from '@angular/core';

@NgModule({
  imports: [
    MatToolbarModule,
    MatButtonModule,
    MatSidenavModule,
    MatIconModule,
    MatCheckboxModule,
    MatListModule,
    MatRadioModule,
    MatSelectModule,
    MatTabsModule,
    MatInputModule,
    MatProgressSpinnerModule,
    MatChipsModule,
    MatCardModule,
    MatMenuModule,
    MatTooltipModule,
    MatSnackBarModule,
    MatSlideToggleModule
  ],
  exports: [
    MatToolbarModule,
    MatButtonModule,
    MatSidenavModule,
    MatIconModule,
    MatCheckboxModule,
    MatListModule,
    MatRadioModule,
    MatSelectModule,
    MatTabsModule,
    MatInputModule,
    MatProgressSpinnerModule,
    MatChipsModule,
    MatCardModule,
    MatMenuModule,
    MatTooltipModule,
    MatSnackBarModule,
    MatSlideToggleModule
  ],
})
export class MaterialModule { }
