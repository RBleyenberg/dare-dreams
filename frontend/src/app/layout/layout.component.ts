import { Component } from '@angular/core';
import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { routeAnimations } from './animation/route.animations';

@Component({
  selector: 'app-layout',
  templateUrl: './layout.component.html',
  styleUrls: ['./layout.component.scss'],
  animations: [routeAnimations]
})
export class LayoutComponent {


  constructor(private breakpointObserver: BreakpointObserver) {}

  isHandset$: Observable<boolean> = this.breakpointObserver.observe(Breakpoints.Handset)
    .pipe(
      map(result => result.matches)
    );

}
