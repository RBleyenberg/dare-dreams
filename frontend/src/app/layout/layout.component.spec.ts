import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { MatIconModule, MatListModule, MatSidenavModule, MatToolbarModule } from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';
import { LayoutModule } from '@angular/cdk/layout';
import { BrowserAnimationsModule, NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ConfigureFn, configureTests } from '../../test-config.helper';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { LayoutComponent } from './layout.component';

describe('LayoutComponent', () => {

  let fixture: ComponentFixture<LayoutComponent>;
  let component: LayoutComponent;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [LayoutComponent],
          imports: [
            NoopAnimationsModule,
            RouterTestingModule,
            BrowserAnimationsModule,
            MatToolbarModule,
            MatIconModule,
            MatListModule,
            LayoutModule,
            MatSidenavModule
          ],
          schemas: [NO_ERRORS_SCHEMA],
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(LayoutComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it(
    'should create the layout',
    async(() => {
      const app = component;
      expect(app).toBeTruthy();
    })
  );

  it('snaps', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('looks async but is synchronous', <any>fakeAsync((): void => {
    let flag = false;
    setTimeout(() => {
      flag = true;
    }, 100);
    expect(flag).toBe(false);
    tick(50);
    expect(flag).toBe(false);
    tick(50);
    expect(flag).toBe(true);
  }));

});
