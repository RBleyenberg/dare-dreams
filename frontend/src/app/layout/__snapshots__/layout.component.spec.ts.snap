// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent snaps 1`] = `
<app-layout
  breakpointObserver={[Function BreakpointObserver]}
  isHandset$={[Function Observable]}
>
  <mat-sidenav-container
    class="sidenav-container mat-drawer-container mat-sidenav-container"
  >
    
    <div
      class="mat-drawer-backdrop ng-star-inserted"
    />
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
      tabindex="-1"
    />
    <mat-sidenav
      class="sidenav mat-drawer mat-sidenav ng-tns-c8-1 ng-trigger ng-trigger-transform ng-tns-c6-0 mat-drawer-side mat-sidenav-fixed ng-star-inserted"
      fixedinviewport="true"
      ng-reflect-fixed-in-viewport="true"
      ng-reflect-mode="side"
      ng-reflect-opened="true"
      role="navigation"
      style="top:0px;bottom:0px;transform:translate3d(0, 0, 0);visibility:visible;0:top;1:bottom;2:transform;3:visibility;"
      tabindex="-1"
    >
      <mat-toolbar
        class="mat-toolbar ng-tns-c6-0 mat-primary mat-toolbar-single-row"
        color="primary"
        ng-reflect-color="primary"
      >
        Menu
      </mat-toolbar>
      <mat-nav-list
        class="mat-nav-list ng-tns-c6-0"
        role="navigation"
      >
        <a
          class="mat-list-item ng-tns-c6-0"
          href="/articles"
          mat-list-item=""
          ng-reflect-router-link="articles"
          routerlink="articles"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="false"
              ng-reflect-trigger="http://localhost/articles"
            />
            <div
              class="mat-list-text"
            />
            Artikelen
          </div>
        </a>
        <a
          class="mat-list-item ng-tns-c6-0"
          href="/relations"
          mat-list-item=""
          ng-reflect-router-link="relations"
          routerlink="relations"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="false"
              ng-reflect-trigger="http://localhost/relations"
            />
            <div
              class="mat-list-text"
            />
            Relaties
          </div>
        </a>
        <a
          class="mat-list-item ng-tns-c6-0"
          href="/relatie-type"
          mat-list-item=""
          ng-reflect-router-link="relatie-type"
          routerlink="relatie-type"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="false"
              ng-reflect-trigger="http://localhost/relatie-type"
            />
            <div
              class="mat-list-text"
            />
            Relatie type
          </div>
        </a>
        <a
          class="mat-list-item ng-tns-c6-0"
          href="/valutas"
          mat-list-item=""
          ng-reflect-router-link="valutas"
          routerlink="valutas"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="false"
              ng-reflect-trigger="http://localhost/valutas"
            />
            <div
              class="mat-list-text"
            />
            Valuta's
          </div>
        </a>
      </mat-nav-list>
    </mat-sidenav>
    <div
      class="cdk-visually-hidden cdk-focus-trap-anchor"
      tabindex="-1"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content ng-tns-c6-0"
    >
      <mat-toolbar
        class="mat-toolbar ng-tns-c6-0 mat-primary mat-toolbar-single-row"
        color="primary"
        ng-reflect-color="primary"
      >
        
        <span
          class="ng-tns-c6-0"
        >
          frontend
        </span>
      </mat-toolbar>
    </mat-sidenav-content>
    
  </mat-sidenav-container>
</app-layout>
`;
